### DISCUSSION TITLE
Does the [if debug] attribute work for you?

### DISCUSSION THREAD

**Discussion by skoovo** [vlang/v#9198] [General]
Hi folks,

I'm new to V, but I'm liking it so far. I think it has a lot of potential. I was just working my way through the docs, and was trying out the [Attributes](https://github.com/vlang/v/blob/master/doc/docs.md#attributes). In particular, I wanted to try out the `[if debug]` attribute.

However, despite marking a function with the attribute, the function is still called even when no flag is passed. Is this a future feature, or am I just messing up somehow? My test code is nothing more than the example given in the docs. I checked the open issues, but couldn't see any mention of a bug and I'm not a good enough programmer to boldly claim that this is one 😁

Are `[if debug]` attributes working for you? If this is something that others are seeing, I'll open an issue. Thanks.

**Comment by JalonSolov** [comment]
Try with the `-cg` option.

**Reply by skoovo** [reply]
Thanks for the suggestion, but it didn't change the outcome. The function was still called.

**Reply by JalonSolov** [reply]
Hmm... I thought that would have done it.  The only other thing I can think of is to add `-d debug`.  If that still doesn't work, then a bug issue is a good idea (and mention you tried both these).

**Comment by larpon** [comment]
@skoovo have you tried `v -prod ...`?

**Reply by skoovo** [reply]
Updating my version of V solved the issue. Thanks for the suggestion though!

**Comment by spaceface777** [comment]
@skoovo can you make sure you're on the latest V version (2ed73bf as of writing) ?
There was a bugfix regarding this recently.

**Reply by skoovo** [reply]
I just updated to V 0.2.2 15927cb and it works. I was using V 0.2.2 5d653a3. Clearly V development is really active :) Thanks for the tip to update my version!

Out of interest, is there a way to use multiple flags? I've tried a few obvious separators, but it either gave an error or ignored them all.

**Reply by skoovo** [reply]
Actually, just using multiple `-d flag` works.
